lib/shared/src/llm-providers/ollama/completions-client.ts

```diff
@@ -28,83 +25,54 @@ export function createOllamaClient(
         params: OllamaGenerateParams,
         abortController: AbortController
     ): CompletionResponseGenerator {
-        const url = new URL('/api/generate', ollamaOptions.url).href
+        const url = new URL(ollamaOptions.url).href
+        const ollama = new Ollama({ host: url })
+        const model = params.model
+
         const log = logger?.startCompletion(params, url)
         const { signal } = abortController
 
         try {
-            const response = await fetch(url, {
-                method: 'POST',
-                body: JSON.stringify({
-                    ...params,
-                    prompt: await params.prompt.toFilteredString(contextFiltersProvider),
-                }),
-                headers: {
-                    'Content-Type': 'application/json',
-                    Connection: 'keep-alive',
-                },
-                signal,
-            })
+            const prompt = await params.prompt.toFilteredString(contextFiltersProvider)
 
-            if (!response.ok) {
-                const errorResponse = (await response.json()) as OllamaGenerateErrorResponse
-                throw new Error(`ollama generation error: ${errorResponse?.error || 'unknown error'}`)
-            }
+            const res = await ollama.generate({
+                model,
+                prompt,
+                options: params.options,
+                stream: true,
+            })
 
-            if (!response.body) {
-                throw new Error('no response body')
+            const completionResponse: CompletionResponse = {
+                completion: '',
+                stopReason: CompletionStopReason.StreamingChunk,
             }
 
-            const iterableBody = isNodeResponse(response)
-                ? response.body
-                : browserResponseToAsyncIterable(response.body)
-
-            let insertText = ''
-            let stopReason = ''
-            let buffer = ''
+            for await (const line of res) {
+                if (line.response) {
+                    completionResponse.completion += line.response
+                    yield {
+                        completionResponse,
```

Comment:
Should we keep `stopReason` here?
```suggestion
                    yield {
                        completionResponse,
                        stopReason: CompletionStopReason.StreamingChunk
```